### DISCUSSION TITLE
Dependabot Alerts now show vulnerable function calls

### DISCUSSION THREAD

**Discussion by erinhav** [community/community#15010] [Code Security]
## Vulnerable function calls now in Python beta

Dependabot alerts will now use GitHub’s [precise code navigation engine](https://github.blog/2021-12-09-precise-code-navigation-python-code-navigation-pull-requests/) to determine if a repository directly calls a vulnerable function. That information will then be surfaced to developers via the UI for Dependabot alerts.

![image](https://user-images.githubusercontent.com/5788563/163599079-2866f708-41bb-4ceb-9626-cd5612c3d577.png)

### How does it work?

Today, GitHub curates information on vulnerable packages in our Advisory Database. Moving forward, as part of the advisory publication process, GitHub will curate information on affected functions for each source library. Leveraging GitHub’s semantic code graph, we perform static analysis with these functions in order to generate an affected call graph for your repository, which is surfaced in a Dependabot alert.

### Which advisories and ecosystems are supported?

Today, we have details of vulnerable functions for 79 Python advisories from the pip ecosystem. We’ll continue backfilling data on vulnerable functions for Python advisories through the beta, as well as supporting any new Python advisories. In addition, we plan to make it possible for anyone to view affected functions in GitHub’s advisory database, as well as to suggest vulnerable functions via [community contributions](https://github.blog/2022-02-22-github-advisory-database-now-open-to-community-contributions/) in the near future.

### Who can use the feature?

This feature is enabled for supported Dependabot alerts on public repositories, as well as on repositories with GitHub Advanced Security enabled.

To learn more, read our blog post, [here](https://github.blog/2022-04-14-dependabot-alerts-now-surface-if-code-is-calling-vulnerability/)! Let me know what you think about the feature, or if you're interested in joining a user feedback session to share your thoughts more directly.

**Comment by anima908710** [comment]
I want help
